custom/icds_reports/sqldata/exports/dashboard_usage.py

```diff
@@ -58,206 +56,138 @@ def get_role_from_username(self, username):
-                    user_counts[record['username']] += record['count']
-                    # updating the counts as per the index which is the indicator number
-                    user_indicators[record['username']][int(record['indicator']) - 1] = record['count']
-            # accumulating the indicator counts
-            for user in users:
-                if not dashboard_uname_rx.match(user['username']):
-                    continue
-                indicator_count = user_indicators[user['username']]
-                user_sql_location_ids = user['assigned_location_ids']
-                if isinstance(user_sql_location_ids, str):
-                    user_sql_location_ids = [user_sql_location_ids]
-
-                for user_sql_location in user_sql_location_ids:
-                    # getting the location type to look up in matrix
-                    if user_sql_location not in self.sql_locations:
-                        continue
-                    location_type_id = self.sql_locations[user_sql_location]
-                    column_index = self.location_types.index(location_type_id)
-                    user_location_row = None
-                    # iterating and getting the db row from matrix
-                    for row in location_matrix:
-                        if row[2 * column_index] == user_sql_location:
-                            user_location_row = copy.deepcopy(row)
-                            break
-                    user_location_type = self.get_location_type_string_from_location_id(location_type_id)
-
-                    if user_location_row is not None:
-                        if user_location_type == 'state':
-                            user_location_row[3] = ''
-                            user_location_row[5] = ''
-                        if user_location_type == 'district':
-                            user_location_row[5] = ''
-
-                        last_login, logged_in_last_week = self.check_if_date_in_last_week(user['last_login'])
-
-                        excel = [serial_count, user_location_row[1], user_location_row[3],
-                                 user_location_row[5], user['username'].split('@')[0], user_location_type,
-                                 self.get_role_from_username(user['username']),
-                                 self.convert_boolean_to_string(self.agg_list[user_sql_location]),
-                                 last_login,
-                                 self.convert_boolean_to_string(logged_in_last_week),
-                                 user_counts[user['username']]]
-                        excel.extend(indicator_count)
-                        excel_rows.append(excel)
-            loop_counter += 1
+                district_name = self.get_location_name_from_id(record.district_id)
+                if record.user_level == 2:
+                    block_name = ''
+                else:
+                    block_name = self.get_location_name_from_id(record.block_id)
```

Comment:
this can be simplified down to
```
state_name = self.get_location_name_from_id(record.state_id)
district_name = ''
block_name = ''

if record.user_level >1:
    district_name = self.get_location_name_from_id(record.district_id)
if record.user_level >2:
    block_name = self.get_location_name_from_id(record.block_id)

```

Reply:
changed